### ISSUE TITLE
Name searches are bad at matching nicknames.

### ISSUE THREAD

**Issue by aaronrudkin** [JeffreyBLewis/WebVoteView#65]
Search Jimmy Carter. No result, because he's "James Earl Carter".

Some other examples
- Dan Sullivan
- Tom Carper
- Chris Coons
- Mike Crapo
- Jim Risch
- Ben Cardin
- Ed Markey
- Thad Cochrane (4th result)
- Jim Inhofe
- Ron Wyden (3rd result)
- Pat Leahy (he seems to go by both Patrick and Pat and we get him as 2nd result so it's OK)
- Mike Enzi

And some names I thought of while looking through the Senate:
- Ron Paul
- Newt Gingrich

The scoring algorithm is good enough that you can get a bunch of these guys through searching their last names, but no one should be left out because we search for what they actually go by. How do we want to solve this? Add () paren names to the full names? Store a lemmatized version of the names using a nickname database? Do nothing?

**Comment by adamboche** [comment]
Yep. The bioguide has nicknames for a bunch of people, but evidently many
are left out. I agree we should address this. Storing an additional field
like you suggest seems like an especially good idea to me.

**Comment by lukesonnet** [comment]
I should also note that if you search Lincoln you don't get Abraham. For member search I think it would be prudent to try and blend the score of name search, recency, importance of seat held (president, Senate, house), and time spent in office.

**Comment by aaronrudkin** [comment]
I do blend recency and name score, but I don't incorporate importance of seat held or time spent in office. I think name score still has to be the dominant thing (for one thing, for speed reasons it's far easier to reduce to a maximum set by name and then filter from there). I agree Abraham Lincoln should obviously come first for "Lincoln".

**Comment by aaronrudkin** [comment]
This is fixed. All of the above examples resolve to the right people. I used a nickname list I had kicking around, and basically do a pseudo-lemmatization of the search results and search request to their shortest nickname variant. Whichever score is best between the raw score and the nickname score is kept. The nickname list is in model/nicknames.json -- it should be easy to add to if you notice something missing.  

See also my edits to the above comment on Lincoln for the rest of Luke's suggestions, which I also added.
